pkg/collector/corechecks/cluster/kubernetes_apiserver.go

```diff
@@ -122,6 +180,45 @@ func (k *KubeASCheck) parseComponentStatus(sender aggregator.Sender, componentsS
 	}
 	return nil
 }
+
+func (k *KubeASCheck) aggregateEvents(sender aggregator.Sender, events []*v1.Event, modified bool) error {
+	eventsByObject := make(map[string]*kubernetesEventBundle)
+	filteredByType := make(map[string]int)
+
+ITER_EVENTS:
+	for _, event := range events {
+		for _, action := range k.instance.FilteredEventType {
+			if *event.Reason == action {
+				filteredByType[action] = filteredByType[action] + 1
+				continue ITER_EVENTS
+			}
+		}
+		bundle, found := eventsByObject[*event.InvolvedObject.Uid]
+		if found == false {
+			bundle = newKubernetesEventBundler(*event.InvolvedObject.Uid, *event.Source.Component)
```

Comment:
are these pointers already tested as non `nil` in the apiclient, or should we check here before dereferencing?

Reply:
This one is not nil indeed http://kubernetes.io/docs/user-guide/identifiers#uids